plugins/woocommerce/client/blocks/assets/js/blocks/product-collection/edit/inspector-controls/taxonomy-controls/index.tsx

```diff
@@ -36,66 +40,125 @@ export const useTaxonomies = (): Taxonomy[] => {
 	}, [ taxonomies ] );
 };
 
-function TaxonomyControls( {
+/**
+ * Normalize the name so first letter of every word is capitalized.
+ */
+const normalizeName = ( name: string | undefined | null ) => {
+	if ( ! name ) {
+		return '';
+	}
+
+	return name
+		.split( ' ' )
+		.map( ( word ) => word.charAt( 0 ).toUpperCase() + word.slice( 1 ) )
+		.join( ' ' );
+};
+
+/**
+ * Shared hook for taxonomy control logic - filters taxonomies based on context and provides common handlers.
+ */
+function useTaxonomyControls( {
 	setQueryAttribute,
 	trackInteraction,
 	query,
-}: QueryControlProps ) {
+	collection,
+	isFiltersPanel,
+}: QueryControlProps & { collection: string | undefined } & {
+	isFiltersPanel?: boolean;
+} ) {
 	const { taxQuery } = query;
-
 	const taxonomies = useTaxonomies();
-	if ( ! taxonomies || taxonomies.length === 0 ) {
-		return null;
-	}
 
-	/**
-	 * Normalize the name so first letter of every word is capitalized.
-	 */
-	const normalizeName = ( name: string | undefined | null ) => {
-		if ( ! name ) {
-			return '';
+	const filteredTaxonomies = useMemo( () => {
+		if ( ! taxonomies || taxonomies.length === 0 ) {
+			return [];
 		}
 
-		return name
-			.split( ' ' )
-			.map( ( word ) => word.charAt( 0 ).toUpperCase() + word.slice( 1 ) )
-			.join( ' ' );
+		if ( collection === CoreCollectionNames.BY_CATEGORY ) {
+			return taxonomies.filter( ( taxonomy ) =>
+				isFiltersPanel
+					? taxonomy.slug !== 'product_cat'
+					: taxonomy.slug === 'product_cat'
```

Comment:
Let's please add a comment explaining this logic for others who may look into here in future. E.g.

```suggestion
                                 // If it's in filter panel, we want to show everything BUT the category control.
                                 // Otherwise, it's a collection specific filter and we want to show ONLY the category control.
				isFiltersPanel
					? taxonomy.slug !== 'product_cat'
					: taxonomy.slug === 'product_cat'
```